packages/circuit-ui/components/NotificationBanner/NotificationBanner.tsx

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright 2019, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { HTMLProps, MouseEvent, KeyboardEvent } from 'react';
+import { css } from '@emotion/core';
+import { Theme } from '@sumup/design-tokens';
+import { Dispatch as TrackingProps } from '@sumup/collector';
+
+import Button, { ButtonProps } from '../Button';
+import styled, { StyleProps } from '../../styles/styled';
+import { spacing } from '../../styles/style-mixins';
+import Headline from '../Headline';
+import Body from '../Body';
+import Image from '../Image';
+import CloseButton from '../CloseButton';
+import { BUTTON_BORDER_WIDTH } from '../Button/Button';
+
+type Action = ButtonProps & {
+  variant: 'primary' | 'tertiary';
+};
+
+type CloseProps =
+  | {
+      /**
+       * Renders a close button inside the tag and calls the provided function
+       * when the button is clicked.
+       */
+      onClose: (event: MouseEvent | KeyboardEvent) => void;
+      /**
+       * Text label for the remove icon for screen readers.
+       * Important for accessibility.
+       */
+      closeButtonLabel: string;
+    }
+  | { onClose?: never; closeButtonLabel?: never };
+
+interface BaseProps extends Omit<HTMLProps<HTMLDivElement>, 'action'> {
+  /**
+   * Use the `system` variant for system notification use cases,
+   * otherwise, use the `promotional`
+   * variant for promotional notification use cases.
+   */
+  variant?: 'system' | 'promotional';
+  /**
+   * The source URL of the image.
+   */
+  src?: string;
+  /**
+   * Alt text for the image.
+   */
+  alt?: string;
```

Comment:
I would combine these into an object prop. This has two advantages: the `alt` prop can be required and additional props can be passed to the image component (e.g. `srcset`, `sizes`).

```suggestion
  image?: {
    /**
     * The source URL of the image.
     */
    src: string;
    /**
     * Alt text for the image.
     */
    alt?: string;
  }
```